UnityProject/Assets/Scripts/Mobs/AI/MobPathfinderV2.cs

```diff
@@ -0,0 +1,444 @@
+				return;
+			}
+
+			//	debugPath = path;
+			StartCoroutine(PerformFollowPath(path));
+		}
+
+		IEnumerator PerformFollowPath(List<Node> path)
+		{
+			int node = 1;
+
+			while (node < path.Count)
+			{
+				yield return WaitFor.EndOfFrame;
+				if (health.IsDead)
+				{
+					yield break;
+				}
+
+				if (!movingToTile)
+				{
+					if (TickDelay.Approx(0) == false)
+					{
+						yield return WaitFor.Seconds(TickDelay);
+					}
+
+					var dir = path[node].position - Vector2Int.RoundToInt(transform.localPosition);
+					if (!registerTile.Matrix.IsPassableAtOneMatrixOneTile(registerTile.LocalPositionServer + (Vector3Int)dir, true, context: gameObject))
+					{
+						var dC = registerTile.Matrix.GetFirst<DoorController>(
+							registerTile.LocalPositionServer + (Vector3Int)dir, true);
+						if (dC != null)
+						{
+							dC.MobTryOpen(gameObject);
+							yield return WaitFor.Seconds(1f);
+						}
+						else
+						{
+							ResetMovingValues();
+							FollowCompleted();
+							Logger.Log("Path following timed out. Something must be in the way", Category.Movement);
+							yield break;
+						}
+					}
+
+					if (Rotatable != null)
+					{
+						Rotatable.SetFaceDirectionLocalVector(dir);
+					}
+
+					Movement.TryTilePush(dir, null);
+					movingToTile = true;
+				}
+				else
+				{
+					if (arrivedAtTile)
+					{
+						movingToTile = false;
+						arrivedAtTile = false;
+						timeOut = 0f;
+						node++;
+					}
+					else
+					{
+						//Mob has 5 seconds to get to the next tile
+						//or the AI should do something else
+						timeOut += MobController.UpdateTimeInterval;
+						if (timeOut > 5f)
+						{
+							ResetMovingValues();
+							Logger.Log("Path following timed out. Something must be in the way", Category.Movement);
+							FollowCompleted();
+							yield break;
+						}
+					}
+				}
+
+				yield return WaitFor.EndOfFrame;
+			}
+
+			yield return WaitFor.EndOfFrame;
+
+			ResetMovingValues();
+			FollowCompleted();
+		}
+
+		private void ResetMovingValues()
+		{
+			movingToTile = false;
+			arrivedAtTile = false;
+			timeOut = 0f;
+		}
+
+		/// <summary>
+		/// This is called whenever a follow path has been completed
+		/// It is also called on a path follow timeout or if something has
+		/// blocked the path and the NPC cannot go any further
+		/// </summary>
+		protected virtual void FollowCompleted()
+		{
+		}
+
+		/// <summary>
+		/// This method is called if something has moved into the path
+		/// and has been blocking the agents path forward for
+		/// more then 5 seconds. PathFinder will go back to idle when
+		/// this is called
+		/// </summary>
+		protected virtual void PathMoveTimedOut()
+		{
+		}
+
+		protected virtual void OnTileReached(Vector3Int oldLocalPos, Vector3Int newLocalPos)
+		{
+			if (movingToTile && !arrivedAtTile)
+			{
+				arrivedAtTile = true;
+			}
+		}
+
+		private void RefreshNodeType(Node node)
```

Comment:
why is this here?

Reply:
idk

Reply:
isn't it meant to be in the meta data node system?

Reply:
```asm
include io.h
data segment
  character db 'idk',0
  data ends
code segment
  start :
  repnz db
  jmp start 
data ends
end start
```